corehq/apps/enterprise/enterprise.py

```diff
@@ -635,3 +634,148 @@ def rows_for_domain(self, domain_obj):
+        return rows
+
+    @property
+    def total(self):
+        total_out_of_date_rows = 0
+        total_last_builds = 0
+
+        for domain in self.account.get_domains():
+            domain_out_of_date_rows, domain_last_builds = self.total_for_domain(domain)
+            total_out_of_date_rows += domain_out_of_date_rows
+            total_last_builds += domain_last_builds
+
+        total_up_to_date = total_last_builds - total_out_of_date_rows
+
+        return _format_percentage_for_enterprise_tile(total_up_to_date, total_last_builds)
+
+    def rows_for_domain(self, domain):
+        rows = []
+        app_name_by_id = {}
+        apps = get_app_ids_in_domain(domain)
+
+        for user, build, latest_version in self._get_user_builds(domain, apps):
+            if is_out_of_date(str(build['build_version']), str(latest_version)):
+                app_id = build['app_id']
+                if app_id not in app_name_by_id:
+                    app_name_by_id[app_id] = Application.get_db().get(app_id).get('name')
+                rows.append([
+                    user['username'],
+                    domain,
+                    app_name_by_id[app_id],
+                    latest_version,
+                    build['build_version'],
+                    self.format_date(DateTimeProperty.deserialize(build['build_version_date'])),
+                ])
+
+        return rows
+
+    def total_for_domain(self, domain):
+        apps = get_app_ids_in_domain(domain)
+        total_last_builds = 0
+        total_out_of_date = 0
+
+        for user, build, latest_version in self._get_user_builds(domain, apps):
+            total_last_builds += 1
+            if is_out_of_date(str(build['build_version']), str(latest_version)):
+                total_out_of_date += 1
+
+        return total_out_of_date, total_last_builds
+
+    def _get_user_builds(self, domain, apps):
+        user_query = (UserES()
+            .domain(domain)
+            .mobile_users()
+            .source([
+                'username',
+                'reporting_metadata.last_builds',
+            ]))
+        for user in user_query.run().hits:
+            last_builds = user.get('reporting_metadata', {}).get('last_builds', [])
+            for build in last_builds:
+                app_id = build.get('app_id')
+                build_version = build.get('build_version')
+                if app_id not in apps or not build_version:
+                    continue
+                build_version_date = DateTimeProperty.deserialize(build.get('build_version_date'))
+                all_builds = self.get_app_builds(domain, app_id, limit=self.INITIAL_QUERY_LIMIT)
+                latest_version = self.get_latest_build_version_at_time(all_builds, build_version_date)
+                if not latest_version:
+                    all_builds = self.get_app_builds(domain, app_id)
+                    latest_version = self.get_latest_build_version_at_time(all_builds, build_version_date)
```

Comment:
It seems like `get_app_builds` is only used by `get_latest_build_version_at_time`. Rather than just retrying the query here, could that `if not` logic be moved (or removed) if `get_app_builds` is called directly by `get_latest_build_version_at_time`?